microprofile/WFLY-19866-promote-mp-rest-client-to-default.adoc

```diff
@@ -0,0 +1,135 @@
+:idseparator:       -
+
+== Overview
+
+This document is about promoting the http://download.eclipse.org/microprofile/microprofile-rest-client-4.0/microprofile-rest-client-spec-4.0.html[MicroProfile REST Client 4.0]
+from a stability level of `preview` to a stability level of `default`.
+
+As you'll see in the release notes below, not much was added to the MicroProfile REST Client 4.0 API. There are only two
+new methods added to the `RestClientBuilder` API.
+
+=== User Stories
+
+A user wants to use the MicroProfile REST Client version 4.0 or use MicroProfile Platform 7.0.
+
+== Issue Metadata
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19866[WFLY-19866] - Promotion from `preview` to `default`
+* https://issues.redhat.com/browse/WFLY-19589[WFLY-19589] - Integration of MicroProfile REST Client 4.0 into WildFly Preview
+* https://issues.redhat.com/browse/WFLY-19588[WFLY-19588] - MicroProfile Platform 7.0
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly
+
+=== Other Interested Projects
+
+Any project wanting either the MicroProfile REST Client version 4.0 or MicroProfile Platform 7.0.
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+The MicroProfile REST Client API and the RESTEasy MicroProfile implementation must be upgraded to be compliant with the
+https://download.eclipse.org/microprofile/microprofile-7.0/microprofile-spec-7.0.html[MicroProfile 7.0 Specification].
+
+There are no specific integration requirements as this is about upgrading an API and implementation.
+
+=== Changed requirements
+
+The only change here is migrating from the MicroProfile REST Client 3.0 API to the MicroProfile REST Client 4.0 API and
+RESTEasy MicroProfile 2.1 to RESTEasy MicroProfile 3.0.
+
+=== Non-Requirements
+
+N/A
+
+=== Future Work
+
+N/A
+
+== Backwards Compatibility
+
+There are no backwards compatibility issues. The new API has only added two new methods to the `RestClientBuilder` API.
+
+=== Default Configuration
+
+There are currently no configuration options for the integration and no new ones will be added.
+
+=== Importing Existing Configuration
+
+There is no issue using a current WildFly configuration with this change.
+
+=== Deployments
+
+There is no incompatible changes to the API or implementation. Deployments will simply see the newer versions available
+on their class path.
+
+=== Interoperability
+
+N/A
+
+== Implementation Plan
+
+The implementation plan is to simply upgrade versions of the required components.
+
+One note is that TCK POM configuration needed some changes. However, that is only for testing.
+
+== Admin Clients
+
+This change has no affect on admin clients. There are no model changes.
+
+== Security Considerations
+
+There would be no changes to security considerations for this upgrade.
+
+[[test_plan]]
+== Test Plan
```

Comment:
@jamezp Can you please update this section based on [the template](https://github.com/wildfly/wildfly-proposals/blob/790b15b89999850f151e25190266a44d8e16ab35/design-doc-template.adoc?plain=1#L133)?

Reply:
Ok, if you don't plan to do any manual or miscellaneous tests, you can just describe it in the analysis. Maybe something like `Manual tests are not necessary because TCK covers all changes` if you agree with it.